src/coap_io_contiki.c

```diff
@@ -217,7 +239,13 @@ int
 coap_io_process(coap_context_t *ctx, uint32_t timeout_ms) {
   coap_tick_t before, now;
 
+  if (timeout_ms != COAP_IO_NO_WAIT) {
+    coap_log_err("coap_io_process must be called with COAP_IO_NO_WAIT\n");
+    return -1;
+  }
+
   coap_ticks(&before);
+  prepare_io(ctx);
   coap_io_do_io(ctx, before);
```

Comment:
I think these 2 lines should be wrapped with
````
PROCESS_CONTEXT_BEGIN(&libcoap_io_process);
PROCESS_CONTEXT_END(&libcoap_io_process);
````
so everything libcoap wise is run under the libcoap_io_process.